### ISSUE TITLE
`array_api_compat.torch.take` with 0-d `indices`?

### ISSUE THREAD

**Issue by mdhaber** [data-apis/array-api-compat#460]
`array_api_compat.torch.take` with 0-d `indices` produces a result with the same number of dimensions as the input.

```python3
from array_api_compat import torch as xp
x = xp.asarray([1])
xp.take(x, xp.asarray(0))
# tensor([1])
```

For non-wrapped `torch.take` (and all other array libraries) the output has one fewer dimension than the input.

```python3
import torch
x = torch.tensor([1])
torch.take(x, torch.asarray(0))
tensor(1)

from array_api_compat import numpy as xp
x = xp.asarray([1])
xp.take(x, xp.asarray(0))
# np.int64(1)
```

Technically, the standard of [`take`](https://data-apis.org/array-api/2022.12/API_specification/generated/array_api.take.html) requires `indices` to have exactly one dimension. 
However, it also says:
> Conceptually, `take(x, indices, axis=3)` is equivalent to `x[:,:,:,indices,...]`;...

This is up for discussion in data-apis/array-api#1010. 

But I'd think that either all libraries should follow the letter of the standard as closely as possible in this situation ("The output array must have the same rank (i.e., number of dimensions) as `x`") or `array_api_compat.torch.take` should be consistent with `torch.take`.

**Comment by ev-br** [comment]
The dreaded _spec mandates behavior more general than what torch implements_ strikes back again:

- the spec requires the `axis` argument and negative indices;
- [`torch.take`](https://docs.pytorch.org/docs/2.13/generated/torch.take.html) does support either of these;
- [the compat layer](https://github.com/data-apis/array-api-compat/blob/main/src/array_api_compat/torch/_aliases.py#L840-L851)  works around it.
- the 0D behavior just falls out of the current workaround.

The 0D behavior is indeed not mandated by the spec, so what compat layer does is an extension to both torch and the spec behaviors.   
A possible argument could be consistency with other major libraries in how they extend the spec: 
https://github.com/data-apis/array-api/issues/1010
